src/main/java/org/veupathdb/service/eda/ss/model/reducer/ancestor/AncestorMapperFactory.java

```diff
@@ -0,0 +1,52 @@
+package org.veupathdb.service.eda.ss.model.reducer.ancestor;
+
+import org.veupathdb.service.eda.ss.model.Entity;
+import org.veupathdb.service.eda.ss.model.Study;
+import org.veupathdb.service.eda.ss.model.variable.binary.AncestorDeserializer;
+import org.veupathdb.service.eda.ss.model.variable.binary.BinaryFilesManager;
+import org.veupathdb.service.eda.ss.model.variable.binary.ListConverter;
+import org.veupathdb.service.eda.ss.model.variable.binary.LongValueConverter;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Iterator;
+
+public class AncestorMapperFactory {
```

Comment:
Needs class level docs.  I don't know about the name AncestorMapperFactory.  I'd think such a factory would produce an AncestorMapper.  This seems to be more like an IdIndexStreamEntityConverter ??

Reply:
Yeah, this went through evolution where it used to return an interface that implemented `AncestorMapper` but it seemed like excessive indirection.